examples/scaling_example/scaling.c

```diff
@@ -230,27 +233,90 @@ packet_handler(struct rte_mbuf *pkt, struct onvm_pkt_meta *meta, __attribute__((
                 data = (void *)rte_malloc("nf_state_data", sizeof(uint16_t), 0);
                 *(uint16_t *)data = destination;
                 /* Get the filled in scale struct by inheriting parent properties */
-                scale_info = onvm_nflib_inherit_parent_config(nf_info, data);
-                scale_info->service_id = destination;
-                scale_info->pkt_func = &packet_handler_child;
+                scale_info = onvm_nflib_inherit_parent_config(nf, data);
+                scale_info->nf_init_cfg->service_id = destination;
+                scale_info->functions.pkt_handler = &packet_handler_child;
                 if (use_shared_cpu_core_allocation)
-                        scale_info->flags = ONVM_SET_BIT(0, SHARE_CORE_BIT);
+                        scale_info->nf_init_cfg->init_options = ONVM_SET_BIT(0, SHARE_CORE_BIT);
                 /* Spawn the child */
                 if (onvm_nflib_scale(scale_info) == 0)
                         RTE_LOG(INFO, APP, "Spawning child SID %u; with packet_handler_child packet function\n",
-                                scale_info->service_id);
+                                scale_info->nf_init_cfg->service_id);
                 else
                         rte_exit(EXIT_FAILURE, "Can't initialize the first child!\n");
         }
 
-        meta->destination = nf_info->service_id;
+        meta->destination = nf->service_id;
         meta->action = ONVM_NF_ACTION_TONF;
 
         return 0;
 }
 
+struct child_spawn_info {
+        struct onvm_nf_init_cfg *child_cfg;
+        struct onvm_nf *parent;
+};
+
 static void
-run_advanced_rings(struct onvm_nf_context *nf_context) {
+run_advanced_rings(struct onvm_nf_local_ctx *nf_local_ctx) {
+        struct onvm_nf *nf;
+        struct onvm_configuration *onvm_config;
+        pthread_t nf_thread[num_children];
+        int i;
+
+        nf = nf_local_ctx->nf;
+        onvm_config = onvm_nflib_get_onvm_config();
+        ONVM_ENABLE_SHARED_CPU = onvm_config->flags.ONVM_ENABLE_SHARED_CPU;
+
+        for (i = 0; i < num_children; i++) {
+                struct onvm_nf_init_cfg *child_cfg;
+                child_cfg = onvm_nflib_init_nf_init_cfg(nf->tag);
+                /* Prepare state data for the child */
+                child_cfg->service_id = nf->service_id;
+                struct child_spawn_info *child_data = malloc(sizeof(struct child_spawn_info));
+                child_data->child_cfg = child_cfg;
+                child_data->parent = nf;
+                /* Increment the children count because we want the ovnm_nflib to handle children termination */
+                rte_atomic16_inc(&nf->thread_info.children_cnt);
```

Comment:
@nks5295 what are we thinking for advanced rings thread managing here? 

We could either use the same concept as we use internally, or we could just declare a global atomic termination flag and just nuke everything when signal is received. 

I'm a fan of the atomic global as we get to showcase how the NF specific signal handler is supposed to be used too.